components/calendar/src/julian.rs

```diff
@@ -0,0 +1,231 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains types and implementations for the Julian calendar
+
+#[derive(Copy, Clone, Debug, Default)]
+// The Julian calendar
+pub struct Julian;
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+// The inner date type used for representing Date<Georgian>
+pub struct JulianDateInner(IsoDateInner);
```

Comment:
I wonder if we should have an independent Inner type so that it's not easy to accidentally mix ISO and Julian values. We would need JulianDay/Month/Year types as well. The purpose of the type safety was to make mistakes where algorithms are mixed impossible.

I don't think we should do this _now_, but it's a thing we can do after the offset/until functions have been abstracted away.

I suspect we can have internal `Day<Calendar>`, `Month<Calendar>`, etc types and a shared `BasicDateInner<Calendar>` type that has the trait magic.